src/plugins/saved_objects_management/server/routes/find.ts

```diff
@@ -77,22 +79,23 @@ export const registerFindRoute = (
         searchFields: [...searchFields],
       });
 
-      const enhancedSavedObjects = findResponse.saved_objects
-        .map((so) => injectMetaAttributes(so, managementService))
-        .map((obj) => {
+      const savedObjects = findResponse.saved_objects.map(toSavedObjectWithMeta);
+
+      const response: FindResponseHTTPV1 = {
+        saved_objects: savedObjects.map((so) => {
+          const obj = injectMetaAttributes(so, managementService);
           const result = { ...obj, attributes: {} as Record<string, any> };
           for (const field of includedFields) {
-            result.attributes[field] = obj.attributes[field];
+            result.attributes[field] = (obj.attributes as Record<string, any>)[field];
```

Comment:
The need to cast here is a little nasty, although understandable.
Could we improve this a bit with:
```suggestion
            result.attributes[field] = (obj.attributes as Record<string, unknown>)[field];
```
?